feat: add clang-tidy for pika

ForestLH opened the pull request:
fix https://github.com/OpenAtomFoundation/pika/issues/1303

**WIP**
打算分为3个部分，
1. 成功运行clang-tidy，以及修改部分代码**(已完成)**
2. 打开clang-tidy的google-runtime-int**(未完成)**
      将long或者是long long换成int64_t，还有对应的unsigned long long
3. 打开clang-tidy的readability-identifier-naming，修改pika的命名规范**(未完成)**

使用方法，进入输出编译结果的文件夹中，例如output
1. `cd output`
2. `make clang-tidy`
*最后，pika仍然有部分会编译失败的代码，例如src/storage/benchmark，所以我在run_clang_tidy.py里面留了一个TODO，目前手动排除了对这些文件的检查*

ForestLH committed:
feature:add clang-tidy for pika

There are several conflicting files left that need to be manually modified. See run_clang_tidy.py.And disabled some clang-tidy checks, eg, use int64_t instead of long or long long, variable naming.

Signed-off-by: Hao Lee <1838249551@qq.com>

ForestLH committed:
style:Make the code standard according to clang-tidy

Signed-off-by: Hao Lee <1838249551@qq.com>

ForestLH committed:
feature:WIP modified clang-tidy file

Signed-off-by: Hao Lee <1838249551@qq.com>

ForestLH committed:
feature:disable compile command clang-tidy check

Signed-off-by: Hao Lee <1838249551@qq.com>

AlexStocks commented on CMakeLists.txt:
```diff
@@ -89,6 +89,29 @@ if (${AUTOCONF} MATCHES AUTOCONF-NOTFOUND)
 endif()
 
+set(CLANG_SEARCH_PATH "/usr/local/bin" "/usr/bin" "/usr/local/opt/llvm/bin" "/usr/local/opt/llvm@12/bin"
+        "/usr/local/Cellar/llvm/12.0.1/bin" "/opt/homebrew/opt/llvm@12/bin/")
```
设置这些路径不合适吧？这就是 mac 的路径吧

infdahai replied:
这里路径删掉（尤其是后面与llvm相关的路径）

AlexStocks commented on include/pika_binlog.h:
```diff
@@ -123,10 +124,6 @@ class Binlog {
   std::atomic<bool> binlog_io_error_;
   // Not use
   // int32_t retry_;
-
```
这里不应该动

ForestLH replied:
这个地方使用delete放到了public里面

AlexStocks commented on include/pika_bit.h:
```diff
@@ -17,90 +17,90 @@
  private:
   std::string key_;
   int64_t bit_offset_ = -1;
-  virtual void Clear() {
```
卧槽，太疯狂了，下面这么多 virtual 都删了？

ForestLH replied:
是的，clang-tidy建议使用override而不是virtual，确实很多，这个改动特别大，辛苦review了

AlexStocks replied:
我查了，就算是建议使用 override，在子类里面加上就行了，可以显著提升可读性。但也并不意味着需要把 父类 中的 virtual 删掉。

lqxhub replied:
> 我查了，就算是建议使用 override，在子类里面加上就行了，可以显著提升可读性。但也并不意味着需要把 父类 中的 virtual 删掉。

我感觉clang-tidy的意思是, 子类不用写virtual, 但是要写override.提高可读性. override只是一个检查做用, 声明子类的这个函数是重写了父类的.  父类的virtual不能省略, 省略了, 子类就不能重写父类的函数了

ForestLH replied:
对的，父类的virtual需要保留，子类只写override

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -94,7 +94,7 @@ Status ClientThread::Close(const std::string& ip, const int port) {
 
 Status ClientThread::ProcessConnectStatus(NetFiredEvent* pfe, int* should_close) {
-  if (pfe->mask & kErrorEvent) {
+  if ((pfe->mask & kErrorEvent) != 0) {
```
这里明显不用改动。不行就把相关规则删了。

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -105,7 +105,7 @@ Status ClientThread::ProcessConnectStatus(NetFiredEvent* pfe, int* should_close)
     return Status::Corruption("Get Socket opt failed");
   }
-  if (val) {
+  if (val != 0) {
```
这里明显不用改动。不行就把相关规则删了。

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -224,10 +226,10 @@ void ClientThread::DoCronTask() {
       close(conn->fd());
       handle_->FdTimeoutHandle(conn->fd(), conn->ip_port());
-      if (ipport_conns_.count(conn->ip_port())) {
+      if (ipport_conns_.count(conn->ip_port()) != 0U) {
```
这里明显不用改动。不行就把相关规则删了。

ForestLH replied:
好的，雨哥，你再看看其他类型的，这个确实有点儿多

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -224,10 +226,10 @@ void ClientThread::DoCronTask() {
         ipport_conns_.erase(conn->ip_port());
       }
-      if (connecting_fds_.count(conn->fd())) {
+      if (connecting_fds_.count(conn->fd()) != 0U) {
```
这里明显不用改动。不行就把相关规则删了。

AlexStocks commented on tools/txt_to_pika/txt_to_pika.cc:
```diff
@@ -59,11 +59,12 @@ int main(int argc, char** argv) {
 
-  for (int i = 0; i < thread_num; i++) {
+  senders.reserve(thread_num);
+for (int i = 0; i < thread_num; i++) {
```
这个都没有对齐

ForestLH replied:
好的，我有的忘记format一下了

infdahai commented on include/pika_command.h:
```diff
@@ -379,17 +380,17 @@ class Cmd : public std::enable_shared_from_this<Cmd> {
     ProcessArg(std::shared_ptr<Partition> _partition, std::shared_ptr<SyncMasterPartition> _sync_partition,
                HintKeys _hint_keys)
-        : partition(_partition), sync_partition(_sync_partition), hint_keys(_hint_keys) {}
+        : partition(std::move(std::move(_partition))), sync_partition(std::move(std::move(_sync_partition))), hint_keys(std::move(std::move(_hint_keys))) {}
```
这里为啥两个std::move()

AlexStocks replied:
我也比较疑惑，为啥两个 move

ForestLH replied:
这里是自动fix的时候出的错，我没看到这里

infdahai commented on include/pika_consensus.h:
```diff
@@ -5,6 +5,10 @@
 
+#include <utility>
+
+#include <utility>
```
去除重复

infdahai commented on include/pika_consensus.h:
```diff
@@ -14,15 +18,15 @@
 class Context {
  public:
-  Context(const std::string path);
+  Context(std::string path);
```
这里应该是建议用std::move吧。

infdahai commented on include/pika_consensus.h:
```diff
@@ -35,13 +39,13 @@ class Context {
     return tmp_stream.str();
   }
+  // No copying allowed;
+  Context(const Context&) = delete;
```
这里可以提供 nocopyable类用于禁止拷贝

infdahai commented on include/pika_consensus.h:
```diff
@@ -69,9 +73,9 @@ class SyncProgress {
+    LogItem(const LogOffset& _offset, std::shared_ptr<Cmd> _cmd_ptr, std::shared_ptr<PikaClientConn> _conn_ptr,
             std::shared_ptr<std::string> _resp_ptr)
-        : offset(_offset), cmd_ptr(_cmd_ptr), conn_ptr(_conn_ptr), resp_ptr(_resp_ptr) {}
+        : offset(_offset), cmd_ptr(std::move(std::move(_cmd_ptr))), conn_ptr(std::move(std::move(_conn_ptr))), resp_ptr(std::move(std::move(_resp_ptr))) {}
```
同上面的std::move

infdahai commented on include/pika_consensus.h:
```diff
@@ -153,7 +157,7 @@ class ConsensusCoordinator {
   // redis parser cb
   struct CmdPtrArg {
-    CmdPtrArg(std::shared_ptr<Cmd> ptr) : cmd_ptr(ptr) {}
+    CmdPtrArg(std::shared_ptr<Cmd> ptr) : cmd_ptr(std::move(std::move(ptr))) {}
```
同上

infdahai commented on include/pika_define.h:
```diff
@@ -200,24 +183,21 @@ const std::string BinlogSyncStateMsg[] = {"NotSync", "ReadFromCache", "ReadFromF
   LogOffset offset_;
   std::string binlog_;
-  BinlogChip(LogOffset offset, std::string binlog) : offset_(offset), binlog_(binlog) {}
+  BinlogChip(const LogOffset& offset, std::string binlog) : offset_(offset), binlog_(std::move(std::move(binlog))) {}
```
同上

infdahai commented on .github/workflows/pika.yml:
```diff
@@ -30,7 +30,7 @@ jobs:
         if: ${{ steps.cache.output.cache-hit != 'true' }}
         run: |
-          sudo apt install autoconf libprotobuf-dev protobuf-compiler -y
+          sudo apt install autoconf libprotobuf-dev protobuf-compiler clang-tidy -y
```
clang-tidy 建议表明下版本。因为clang-tidy-14 和 clang-tidy-16是存在差异的。

infdahai replied:
后面cmake表明的是12，建议一致

AlexStocks replied:
你的意思是，这里直接把 clang-tidy 改为 clang-tidy-14 么？

infdahai replied:
如果根据cmakelists.txt中的改动，应该改成clang-tidy-12

```cmake
find_program(CLANG_TIDY_BIN
        NAMES clang-tidy clang-tidy-12
        HINTS ${CLANG_SEARCH_PATH})
```

AlexStocks replied:
@ForestLH 你看着改下吧

AlexStocks commented on include/pika_bit.h:
```diff
@@ -17,90 +17,90 @@
-  virtual void Merge(){};
-  virtual Cmd* Clone() override { return new BitPosCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
```
关键字 override 和 后面的 `{` 之间加个空格吧

AlexStocks commented on include/pika_bit.h:
```diff
@@ -17,90 +17,90 @@
-  virtual Cmd* Clone() override { return new BitPosCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
关键字 override 和 后面的 `{` 之间加个空格吧

AlexStocks commented on include/pika_bit.h:
```diff
@@ -109,34 +109,34 @@ class BitPosCmd : public Cmd {
-  virtual void Merge(){};
-  virtual Cmd* Clone() override { return new BitOpCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
```
关键字 override 和 后面的 `{` 之间加个空格吧

AlexStocks commented on include/pika_bit.h:
```diff
@@ -109,34 +109,34 @@ class BitPosCmd : public Cmd {
-  virtual Cmd* Clone() override { return new BitOpCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
关键字 override 和 后面的 `{` 之间加个空格吧

infdahai commented on include/pika_bit.h:
```diff
@@ -17,90 +17,90 @@
-  virtual void Merge(){};
-  virtual Cmd* Clone() override { return new BitSetCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
```
这里为空的函数体都用default吧。

AlexStocks commented on include/pika_geohash.h:
```diff
@@ -60,25 +60,25 @@ typedef enum {
+};
 
-typedef struct {
+using GeoHashRange = struct {
```
这里很奇怪，你上面的 typedef GeoHashBits 没有使用 using，而下面 三个 struct 却是用了 using，是何道理？

AlexStocks commented on include/pika_geohash_helper.h:
```diff
@@ -34,26 +34,26 @@
 GeoHashRadius geohashGetAreasByRadiusWGS84(double longitude, double latitude, double radius_meters);
 GeoHashRadius geohashGetAreasByRadiusMercator(double longitude, double latitude, double radius_meters);
-GeoHashFix52Bits geohashAlign52Bits(const GeoHashBits hash);
+GeoHashFix52Bits geohashAlign52Bits(const GeoHashBits &hash);
```
这里的引用符号 &  是不是跟前面的类型符号连在一起比较合适？

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual void Merge(){};
-  virtual Cmd* Clone() override { return new HGetallCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
```
同上，override 最好跟后面 `{` 之间加一个空格

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual Cmd* Clone() override { return new HGetallCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
同上，override 最好跟后面 `{` 之间加一个空格

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual void Merge(){};
-  virtual Cmd* Clone() override { return new HSetCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
```
同上，override 最好跟后面 `{` 之间加一个空格

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual Cmd* Clone() override { return new HSetCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
同上，override 最好跟后面 `{` 之间加一个空格

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual Cmd* Clone() override { return new HExistsCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
同上，处理下空格问题

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual Cmd* Clone() override { return new HIncrbyCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
同上，处理下空格问题

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual Cmd* Clone() override { return new HIncrbyfloatCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
同上，处理下空格问题

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual Cmd* Clone() override { return new HKeysCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
同上，处理下空格问题

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual Cmd* Clone() override { return new HLenCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
同上，处理下空格问题

AlexStocks commented on include/pika_hash.h:
```diff
@@ -17,277 +17,277 @@
-  virtual Cmd* Clone() override { return new HMgetCmd(*this); }
+  void Do(std::shared_ptr<Partition> partition = nullptr) override;
+  void Split(std::shared_ptr<Partition> partition, const HintKeys& hint_keys) override{};
+  void Merge() override{};
```
后面我不在留comment 了，处理下所有的 override 的空格问题

infdahai commented on CMakeLists.txt:
```diff
@@ -761,3 +784,20 @@ if (ENABLE_IPO)
+add_custom_target(clang-tidy-fix
+    ${CMAKE_CURRENT_SOURCE_DIR}/run_clang_tidy.py
+    -clang-tidy-binary ${CLANG_TIDY_BIN}
+    -p ${CMAKE_BINARY_DIR}
```
这里header-filter建议在.clang-tidy处理，为了可读性

AlexStocks commented on include/pika_repl_client.h:
```diff
@@ -23,34 +24,36 @@
   std::shared_ptr<net::PbConn> conn;
   ReplClientTaskArg(std::shared_ptr<InnerMessage::InnerResponse> _res, std::shared_ptr<net::PbConn> _conn)
-      : res(_res), conn(_conn) {}
+      : res(std::move(std::move(_res))), conn(std::move(std::move(_conn))) {}
```
两个 move 的问题，你处理下吧，下面我看也出现了好多次

ForestLH replied:
好的雨哥，我有空再根据这些问题再改下，改过之后再来review

AlexStocks replied:
我已经处理了

AlexStocks commented on include/pika_repl_client_thread.h:
```diff
@@ -45,7 +45,7 @@ class PikaReplClientThread : public net::ClientThread {
     int CreateWorkerSpecificData(void** data) const override { return 0; }
     int DeleteWorkerSpecificData(void* data) const override { return 0; }
-    void DestConnectFailedHandle(std::string ip_port, std::string reason) const override {}
+    void DestConnectFailedHandle(const std::string &ip_port, const std::string &reason) const override {}
```
`const std::string &` --> `const std::string&`

infdahai commented on include/pika_binlog.h:
```diff
@@ -14,20 +14,18 @@
-using pstd::Status;
 
-std::string NewFileName(const std::string name, const uint32_t current);
+std::string NewFileName(const std::string& name, uint32_t current);
```
对于string,如果要拷贝就是const std::string& ，如果用std::string，初始化就调用std::move()

infdahai commented on .clang-tidy:
```diff
@@ -0,0 +1,117 @@
+            -readability-make-member-function-const,
+            -readability-qualified-auto,
+            -readability-redundant-access-specifiers,
+            -bugprone-exception-escape,
```
避免添加重复文件，在这里添加个参数 https://clang.llvm.org/extra/clang-tidy/checks/readability/duplicate-include.html

infdahai commented on include/pika_geohash.h:
```diff
@@ -32,26 +32,26 @@
 
 /* Limits from EPSG:900913 / EPSG:3785 / OSGEO:41001 */
-#define GEO_LAT_MIN -85.05112878
+#define GEO_LAT_MIN (-85.05112878)
```
这里常量建议用 const double （如果支持constexpr就用这个）
```suggestion
const dobule GEO_LAT_MIN = -85.05112878;
```

AlexStocks replied:
很好的建议，我来改下

infdahai commented on include/pika_geohash.h:
```diff
@@ -32,26 +32,26 @@
 #define HASHISZERO(r) (!(r).bits && !(r).step)
 #define RANGEISZERO(r) (!(r).max && !(r).min)
-#define RANGEPISZERO(r) (r == nullptr || RANGEISZERO(*r))
+#define RANGEPISZERO(r) ((r) == nullptr || RANGEISZERO(*(r)))
```
这里写成函数func f(r)->bool好点，如果要改的话。

AlexStocks replied:
这个可以先不动。

infdahai commented on include/pika_geohash.h:
```diff
@@ -32,26 +32,26 @@
 #define GEO_LONG_MAX 180
 
-typedef enum {
+using GeoDirection = enum {
```
```suggestion
enum GeoDirection {
```

AlexStocks commented on include/pika_rm.h:
```diff
@@ -176,55 +175,55 @@ class PikaReplicaManager {
 
   // Update binlog win and try to send next binlog
-  Status UpdateSyncBinlogStatus(const RmNode& slave, const LogOffset& offset_start, const LogOffset& offset_end);
+  pstd::Status UpdateSyncBinlogStatus(const RmNode& slave, const LogOffset& range_start, const LogOffset& range_end);
```
这里改的不明所以，为啥要把 offset 改为 range？

AlexStocks commented on include/pika_rsync_service.h:
```diff
@@ -10,7 +10,7 @@
 class PikaRsyncService {
  public:
-  PikaRsyncService(const std::string& raw_path, const int port);
+  PikaRsyncService(std::string  raw_path, int port);
```
`const std::string& raw_path` 这里为啥要改为 `std::string`?

AlexStocks commented on include/pika_server.h:
```diff
@@ -161,8 +160,8 @@ class PikaServer {
-  Status AddTableStruct(std::string table_name, uint32_t num);
-  Status DelTableStruct(std::string table_name);
+  pstd::Status AddTableStruct(const std::string &table_name, uint32_t num);
+  pstd::Status DelTableStruct(const std::string &table_name);
```
`const std::string &` 统一改为 `const std::string& ` 吧，下面这种问题，我不再留言了。

AlexStocks commented on include/pika_server.h:
```diff
@@ -172,19 +171,19 @@ class PikaServer {
   void PartitionSetMaxCacheStatisticKeys(uint32_t max_cache_statistic_keys);
   void PartitionSetSmallCompactionThreshold(uint32_t small_compaction_threshold);
-  bool GetTablePartitionBinlogOffset(const std::string& table_name, uint32_t partition_id, BinlogOffset* const boffset);
+  bool GetTablePartitionBinlogOffset(const std::string& table_name, uint32_t partition_id, BinlogOffset* boffset);
```
这里的 const 没必要去除吧？

AlexStocks commented on include/pika_set.h:
```diff
@@ -15,95 +15,95 @@
  public:
-  SScanCmd(const std::string& name, int arity, uint16_t flag) : Cmd(name, arity, flag), pattern_("*"), count_(10) {}
-  virtual std::vector<std::string> current_key() const {
+  SScanCmd(const std::string& name, int arity, uint16_t flag) : Cmd(name, arity, flag), pattern_("*") {}
```
`, count_(10)` 这里删的没道理啊？

AlexStocks commented on include/pika_slave_node.h:
```diff
@@ -21,15 +21,15 @@ struct SyncWinItem {
   }
   explicit SyncWinItem(const LogOffset& offset, std::size_t binlog_size = 0)
-      : offset_(offset), binlog_size_(binlog_size), acked_(false) {}
+      : offset_(offset), binlog_size_(binlog_size) {}
```
`, acked_(false)` 这个何必删除呢？

AlexStocks commented on include/pika_zset.h:
```diff
@@ -77,80 +77,80 @@ class ZScanCmd : public Cmd {
  public:
   ZsetRangebyscoreParentCmd(const std::string& name, int arity, uint16_t flag)
-      : Cmd(name, arity, flag), left_close_(true), right_close_(true), with_scores_(false), offset_(0), count_(-1) {}
+      : Cmd(name, arity, flag) {}
```
这个不能删掉的，这是我特意加上去的。

AlexStocks commented on include/pika_zset.h:
```diff
@@ -357,155 +357,155 @@ class ZsetRangebylexParentCmd : public Cmd {
   ZLexcountCmd(const std::string& name, int arity, uint16_t flag)
-      : Cmd(name, arity, flag), left_close_(true), right_close_(true) {}
-  virtual std::vector<std::string> current_key() const {
+      : Cmd(name, arity, flag) {}
```
初始化列表里变量，后面几个默认赋值，不能删除

ForestLH replied:
这种clang-tidy好像是建议加到定义成员变量的那里

AlexStocks replied:
同意

AlexStocks commented on include/pika_zset.h:
```diff
@@ -357,155 +357,155 @@ class ZsetRangebylexParentCmd : public Cmd {
   ZRemrangebyscoreCmd(const std::string& name, int arity, uint16_t flag)
-      : Cmd(name, arity, flag), left_close_(true), right_close_(true) {}
-  virtual std::vector<std::string> current_key() const {
+      : Cmd(name, arity, flag) {}
```
同上，这里初始化列表里面的几个变量，不能删除。

AlexStocks commented on include/pika_zset.h:
```diff
@@ -357,155 +357,155 @@ class ZsetRangebylexParentCmd : public Cmd {
   ZRemrangebylexCmd(const std::string& name, int arity, uint16_t flag)
-      : Cmd(name, arity, flag), left_close_(true), right_close_(true) {}
-  virtual std::vector<std::string> current_key() const {
+      : Cmd(name, arity, flag) {}
```
同上，这里初始化列表里面的几个变量，不能删除。

AlexStocks commented on src/net/include/client_thread.h:
```diff
@@ -93,7 +93,7 @@ class ClientHandle {
    * DestConnectFailedHandle(...) will run the invoker's logic when socket connect failed
    */
-  virtual void DestConnectFailedHandle(std::string ip_port, std::string reason) const {
+  virtual void DestConnectFailedHandle(const std::string &ip_port, const std::string &reason) const {
```
'const std::string &' --> `const std::string& `

AlexStocks commented on src/net/include/net_conn.h:
```diff
@@ -26,8 +26,12 @@ class Thread;
  public:
-  NetConn(const int fd, const std::string& ip_port, Thread* thread, NetMultiplexer* mpx = nullptr);
+  NetConn(int fd, std::string  ip_port, Thread* thread, NetMultiplexer* mpx = nullptr);
+#ifdef __ENABLE_SSL
```
原代码没有，而你这里加上 ssl 判断，是何道理啊？

AlexStocks commented on src/net/include/net_pubsub.h:
```diff
@@ -52,17 +52,17 @@ class PubSubThread : public Thread {
 
   struct ConnHandle {
-    ConnHandle(std::shared_ptr<NetConn> pc, ReadyState state = kNotReady) : conn(pc), ready_state(state) {}
+    ConnHandle(std::shared_ptr<NetConn> pc, ReadyState state = kNotReady) : conn(std::move(std::move(std::move(std::move(pc))))), ready_state(state) {}
```
四个 move？

AlexStocks commented on src/net/include/net_pubsub.h:
```diff
@@ -96,14 +99,14 @@ class PubSubThread : public Thread {
   std::string channel_;
   std::string message_;
-  int receivers_;
+  int receivers_{-1};
```
这是使用等号吧

AlexStocks commented on src/net/include/net_thread.h:
```diff
@@ -43,15 +49,10 @@ class Thread {
   pstd::Mutex running_mu_;
-  bool running_;
-  pthread_t thread_id_;
+  bool running_{false};
```
跟其他地方保持风格统一，使用等号

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -100,7 +100,7 @@ Status BackendThread::Close(const int fd) {
 
 Status BackendThread::ProcessConnectStatus(NetFiredEvent* pfe, int* should_close) {
-  if (pfe->mask & kErrorEvent) {
+  if ((pfe->mask & kErrorEvent) != 0) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -111,7 +111,7 @@ Status BackendThread::ProcessConnectStatus(NetFiredEvent* pfe, int* should_close
     return Status::Corruption("Get Socket opt failed");
   }
-  if (val) {
+  if (val != 0) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -244,10 +246,10 @@ void BackendThread::DoCronTask() {
       close(conn->fd());
       handle_->FdTimeoutHandle(conn->fd(), conn->ip_port());
-      if (conns_.count(conn->fd())) {
+      if (conns_.count(conn->fd()) != 0U) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -244,10 +246,10 @@ void BackendThread::DoCronTask() {
         conns_.erase(conn->fd());
       }
-      if (connecting_fds_.count(conn->fd())) {
+      if (connecting_fds_.count(conn->fd()) != 0U) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -307,7 +309,7 @@ void BackendThread::NotifyClose(const int fd) {
 
 void BackendThread::ProcessNotifyEvents(const NetFiredEvent* pfe) {
-  if (pfe->mask & kReadable) {
+  if ((pfe->mask & kReadable) != 0) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -320,7 +322,7 @@ void BackendThread::ProcessNotifyEvents(const NetFiredEvent* pfe) {
         if (ti.notify_type() == kNotiWrite) {
           if (conns_.find(fd) == conns_.end()) {
-            // TODO: need clean and notify?
+            // TODO(): need clean and notify?
```
TODO 后面加上括号作甚？

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -411,15 +413,15 @@ void* BackendThread::ThreadMain() {
       }
 
-      if (connecting_fds_.count(pfe->fd)) {
+      if (connecting_fds_.count(pfe->fd) != 0U) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -411,15 +413,15 @@ void* BackendThread::ThreadMain() {
       }
 
-      if (!should_close && (pfe->mask & kWritable) && conn->is_reply()) {
+      if ((should_close == 0) && ((pfe->mask & kWritable) != 0) && conn->is_reply()) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -433,7 +435,7 @@ void* BackendThread::ThreadMain() {
       }
 
-      if (!should_close && (pfe->mask & kReadable)) {
+      if ((should_close == 0) && ((pfe->mask & kReadable) != 0)) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -445,15 +447,15 @@ void* BackendThread::ThreadMain() {
       }
 
-      if ((pfe->mask & kErrorEvent) || should_close) {
+      if (((pfe->mask & kErrorEvent) != 0) || (should_close != 0)) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/backend_thread.cc:
```diff
@@ -445,15 +447,15 @@ void* BackendThread::ThreadMain() {
           conns_.erase(pfe->fd);
           mu_.unlock();
-          if (connecting_fds_.count(conn->fd())) {
+          if (connecting_fds_.count(conn->fd()) != 0U) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -301,7 +303,7 @@ void ClientThread::NotifyWrite(const std::string& ip_port) {
 
 void ClientThread::ProcessNotifyEvents(const NetFiredEvent* pfe) {
-  if (pfe->mask & kReadable) {
+  if ((pfe->mask & kReadable) != 0) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -338,7 +340,7 @@ void ClientThread::ProcessNotifyEvents(const NetFiredEvent* pfe) {
             std::vector<std::string>& msgs = iter->second;
             for (auto& msg : msgs) {
-              if (ipport_conns_[ip_port]->WriteResp(msg)) {
+              if (ipport_conns_[ip_port]->WriteResp(msg) != 0) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -415,15 +417,15 @@ void* ClientThread::ThreadMain() {
       std::shared_ptr<NetConn> conn = iter->second;
 
-      if (connecting_fds_.count(pfe->fd)) {
+      if (connecting_fds_.count(pfe->fd) != 0U) {
```
去掉后面的非零判断

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -415,15 +417,15 @@ void* ClientThread::ThreadMain() {
       }
 
-      if (!should_close && (pfe->mask & kWritable) && conn->is_reply()) {
+      if ((should_close == 0) && ((pfe->mask & kWritable) != 0) && conn->is_reply()) {
```
恢复原状

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -437,7 +439,7 @@ void* ClientThread::ThreadMain() {
       }
 
-      if (!should_close && (pfe->mask & kReadable)) {
+      if ((should_close == 0) && ((pfe->mask & kReadable) != 0)) {
```
恢复原状

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -450,16 +452,16 @@ void* ClientThread::ThreadMain() {
       }
 
-      if ((pfe->mask & kErrorEvent) || should_close) {
+      if (((pfe->mask & kErrorEvent) != 0) || (should_close != 0)) {
```
恢复原状

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -450,16 +452,16 @@ void* ClientThread::ThreadMain() {
           CloseFd(conn);
           fd_conns_.erase(pfe->fd);
-          if (ipport_conns_.count(conn->ip_port())) {
+          if (ipport_conns_.count(conn->ip_port()) != 0U) {
```
恢复原状

AlexStocks commented on src/net/src/client_thread.cc:
```diff
@@ -450,16 +452,16 @@ void* ClientThread::ThreadMain() {
             ipport_conns_.erase(conn->ip_port());
           }
-          if (connecting_fds_.count(conn->fd())) {
+          if (connecting_fds_.count(conn->fd()) != 0U) {
```
恢复原状

AlexStocks commented on src/net/src/holy_thread.cc:
```diff
@@ -79,7 +79,7 @@ int HolyThread::StartThread() {
 
 int HolyThread::StopThread() {
-  if (private_data_) {
+  if (private_data_ != nullptr) {
```
恢复原状。这个源码文件里整体改下吧，其他处不再留言。

AlexStocks commented on src/net/src/http_conn.cc:
```diff
@@ -122,7 +123,7 @@ bool HTTPRequest::ParseHeadLine(const char* data, int line_start, int line_end)
   path_ = url_;
   // Format path
-  if (headers_.count("host") && path_.find(headers_["host"]) != std::string::npos &&
+  if ((headers_.count("host") != 0U) && path_.find(headers_["host"]) != std::string::npos &&
```
非零判断 恢复原状。这个源码文件里整体改下吧，其他处不再留言。

AlexStocks commented on src/net/src/http_conn.cc:
```diff
@@ -265,7 +268,7 @@ HTTPConn::HTTPConn(const int fd, const std::string& ip_port, Thread* thread, std
       // security_(thread->security()),
 #endif
-      handles_(handles) {
+      handles_(std::move(std::move(handles))) {
```
两个 move。

AlexStocks commented on src/net/src/net_cli.cc:
```diff
@@ -121,7 +114,7 @@ Status NetCli::Connect(const std::string& ip, const int port, const std::string&
         }
 
-        if (val) {
+        if (val != 0) {
```
非零判断 恢复原状。

AlexStocks commented on src/net/src/net_cli.cc:
```diff
@@ -135,7 +128,7 @@ Status NetCli::Connect(const std::string& ip, const int port, const std::string&
     struct sockaddr_in laddr;
     socklen_t llen = sizeof(laddr);
-    getsockname(r->sockfd, (struct sockaddr*)&laddr, &llen);
+    getsockname(r->sockfd, reinterpret_cast<struct sockaddr*>(&laddr), &llen);
```
使用 static_cast

AlexStocks commented on src/net/src/net_epoll.cc:
```diff
@@ -40,8 +40,12 @@ int NetEpoll::NetAddEvent(int fd, int mask) {
 
-  if (mask & kReadable) ee.events |= EPOLLIN;
-  if (mask & kWritable) ee.events |= EPOLLOUT;
+  if ((mask & kReadable) != 0) {
```
非零判断 恢复原状。这个源码文件里整体改下吧，其他处不再留言。

AlexStocks commented on src/net/src/net_epoll.cc:
```diff
@@ -67,23 +75,25 @@ int NetEpoll::NetDelEvent(int fd, int) {
 int NetEpoll::NetPoll(int timeout) {
-  int num_events = epoll_wait(multiplexer_, &events_[0], NET_MAX_CLIENTS, timeout);
-  if (num_events <= 0) return 0;
+  int num_events = epoll_wait(multiplexer_, (events_).data(), NET_MAX_CLIENTS, timeout);
```
把 ’&events_[0]‘ 改为 ’(events_).data()‘ 显然不对吧？

AlexStocks commented on src/net/src/net_interfaces.cc:
```diff
@@ -63,15 +63,17 @@ std::string GetDefaultInterface() {
     }
 
-    ifreq = (struct ifreq*)((char*)ifreq + len);
+    ifreq = reinterpret_cast<struct ifreq*>(reinterpret_cast<char*>(ifreq) + len);
```
使用 static_cast 可否？

AlexStocks commented on src/net/src/net_interfaces.cc:
```diff
@@ -111,21 +113,21 @@ std::string GetIpByInterface(const std::string& network_interface) {
 
     if (ifa->ifa_addr->sa_family == AF_INET) {  // Check it is a valid IPv4 address
-      tmpAddrPtr = &((struct sockaddr_in*)ifa->ifa_addr)->sin_addr;
+      tmpAddrPtr = &(reinterpret_cast<struct sockaddr_in*>(ifa->ifa_addr))->sin_addr;
```
使用 static_cast 可否？

AlexStocks commented on src/net/src/net_kqueue.cc:
```diff
@@ -36,22 +36,23 @@ int NetKqueue::NetAddEvent(int fd, int mask) {
   struct kevent change[2];
 
-  if (mask & kReadable) {
+  if ((mask & kReadable) != 0) {
```
整体把 非零 判断进行代码恢复

AlexStocks commented on src/net/src/net_kqueue.cc:
```diff
@@ -60,17 +61,18 @@ int NetKqueue::NetDelEvent(int fd, int mask) {
   }
 
-  if (cnt == 0) return -1;
+  if (cnt == 0) { return -1;
```
这里加上大括号极好，但是代码需要格式化

AlexStocks commented on src/net/src/net_kqueue.cc:
```diff
@@ -85,7 +87,8 @@ int NetKqueue::NetPoll(int timeout) {
 
   int num_events = ::kevent(multiplexer_, nullptr, 0, &events_[0], NET_MAX_CLIENTS, p_timeout);
-  if (num_events <= 0) return 0;
+  if (num_events <= 0) { return 0;
```
这里加上大括号极好，但是代码需要格式化

AlexStocks commented:
统一说几个共性问题吧，留言留不过来了:
1 两个 move 问题；
2 nocopy 问题，可以统一定义如下类，然后继承：

```c++
class noncopyable
{
 protected:
  noncopyable() = default;
  ~noncopyable() = default;

 private:
  noncopyable(const noncopyable&) = delete;
  void operator=(const noncopyable&) = delete;
};
```

3 非零判断，恢复；
4 指针判空，恢复；
<img width="891" alt="image" src="https://github.com/OpenAtomFoundation/pika/assets/7959374/7ba84991-9481-4cc8-ace9-709776df6808">
5 reinterpret_cast 尽量使用 static_cast；
6 如下图，你去掉 return，但是最后一个 `}` 前面加上空格就不对了，这显然格式化错误
![image](https://github.com/OpenAtomFoundation/pika/assets/7959374/76c31338-59fe-4c2b-a2e5-2ff4027f0d03)
7 'auto*' 统一改为 'auto'

AlexStocks commented on src/net/src/net_pubsub.cc:
```diff
@@ -289,18 +289,18 @@ int PubSubThread::UnSubscribe(std::shared_ptr<NetConn> conn, const std::vector<s
     for (auto& channel : pubsub_channel_) {
-      if (pstd::stringmatchlen(channel.first.c_str(), channel.first.size(), pattern.c_str(), pattern.size(), 0)) {
-        if (channel.second.size() != 0) {
+      if (pstd::stringmatchlen(channel.first.c_str(), channel.first.size(), pattern.c_str(), pattern.size(), 0) != 0) {
```
!= 0 去掉

AlexStocks commented on src/pika.cc:
```diff
@@ -63,7 +63,8 @@ static void PikaGlogInit() {
 
 static void daemonize() {
-  if (fork() != 0) exit(0); /* parent exits */
+  if (fork() != 0) { exit(0); /* parent exits */
```
代码格式化

AlexStocks commented on tools/pika-port/pika_port_3/redis_sender.cc:
```diff
@@ -1,17 +1,19 @@
 
 RedisSender::RedisSender(int id, std::string ip, int64_t port, std::string password)
-    : id_(id), cli_(nullptr), ip_(ip), port_(port), password_(password), should_exit_(false), cnt_(0), elements_(0) {
+    : id_(id), cli_(nullptr), ip_(std::move(std::move(ip))), port_(port), password_(std::move(std::move(password))), should_exit_(false), cnt_(0), elements_(0) {
```
重复 move

AlexStocks commented on tools/pika-port/pika_port_3/pika_sender.h:
```diff
@@ -17,7 +17,7 @@ class PikaSender : public net::Thread {
   PikaSender(std::string ip, int64_t port, std::string password);
   virtual ~PikaSender();
-  void LoadKey(const std::string& cmd);
+  void LoadKey(const std::string& key);
```
这个改动没必要

AlexStocks commented on tools/pika-port/pika_port_3/pika_binlog.h:
```diff
@@ -25,13 +25,13 @@
 class Binlog {
  public:
-  Binlog(const std::string& Binlog_path, const int file_size = 100 * 1024 * 1024);
+  Binlog(std::string  Binlog_path, const int file_size = 100 * 1024 * 1024);
```
这行改动恢复，没必要改。

AlexStocks commented on src/pika_geohash.cc:
```diff
@@ -189,22 +191,25 @@ int geohashDecodeType(const GeoHashBits hash, GeoHashArea* area) {
 
 int geohashDecodeAreaToLongLat(const GeoHashArea* area, double* xy) {
-  if (!xy) return 0;
+  if (xy == nullptr) { return 0;
```
格式化下

AlexStocks commented on src/pika_repl_server_thread.cc:
```diff
@@ -14,8 +14,8 @@ extern PikaReplicaManager* g_pika_rm;
       conn_factory_(this),
-      port_(port),
-      serial_(0) {
+      port_(port)
```
把左括号恢复到上一行

AlexStocks commented on src/pstd/include/pstd_slice.h:
```diff
@@ -99,10 +99,13 @@ inline int Slice::compare(const Slice& b) const {
   int r = memcmp(data_, b.data_, min_len);
   if (r == 0) {
-    if (size_ < b.size_)
+    if (size_ < b.size_) { { {
```
这显然改的就不对嘛

AlexStocks commented on src/pstd/include/rsync.h:
```diff
@@ -16,11 +17,11 @@ struct RsyncRemote {
   int kbps;  // speed limit
-  RsyncRemote(const std::string& _host, const int _port, const std::string& _module, const int _kbps)
-      : host(_host), port(_port), module(_module), kbps(_kbps) {}
+  RsyncRemote(std::string  _host, const int _port, std::string  _module, const int _kbps)
```
这改的有点无厘头了

AlexStocks commented on src/pstd/include/rsync.h:
```diff
@@ -16,11 +17,11 @@ struct RsyncRemote {
 };
 
-int StartRsync(const std::string& rsync_path, const std::string& module, const std::string& ip, const int port,
+int StartRsync(const std::string& raw_path, const std::string& module, const std::string& ip, int port,
```
这改的有点无厘头了

AlexStocks reviewed (commented):
modify storage/include/storage

AlexStocks committed:
merge upstream unstable

AlexStocks committed:
delete double std::move

AlexStocks committed:
auto* -> auto

AlexStocks committed:
using noncopyable

AlexStocks committed:
add space for override{}

AlexStocks committed:
add spaces

AlexStocks committed:
delete != 0

AlexStocks committed:
format if-return

AlexStocks committed:
clear inheritage

AlexStocks committed:
format codes

AlexStocks committed:
format codes

AlexStocks committed:
delete == 0

ForestLH committed:
fix:fix build error

Signed-off-by: Hao Lee <1838249551@qq.com>

ForestLH committed:
style:disable check bool implicit convert in clang-tidy

Signed-off-by: Hao Lee <1838249551@qq.com>

ForestLH committed:
Merge pull request #1 from AlexStocks/lihao-clang

modify clang

ForestLH committed:
style:update code style

Signed-off-by: Hao Lee <1838249551@qq.com>

AlexStocks committed:
delete '!= nullptr'

AlexStocks committed:
format if/while

ForestLH committed:
Merge pull request #2 from ForestLH/style/clang

Style/clang

ForestLH committed:
Merge pull request #3 from AlexStocks/lihao-clang

format if/while

AlexStocks committed:
fix ci failure

ForestLH committed:
Merge pull request #4 from AlexStocks/lihao-clang

fix ci failure

ForestLH committed:
fix:build ci error

Signed-off-by: Hao Lee <1838249551@qq.com>

ForestLH committed:
style:The judgment statement should not be an assignment

Signed-off-by: Hao Lee <1838249551@qq.com>

ForestLH committed:
style:format codes

remove redundant extern statement and use default

Signed-off-by: Hao Lee <1838249551@qq.com>

ForestLH committed:
Merge pull request #5 from ForestLH/style/clang

fix:build ci error  and format codes

AlexStocks commented on src/net/include/thread_pool.h:
```diff
@@ -70,6 +65,12 @@ class ThreadPool {
+  /*
+   * No allowed copy and copy assign
+   */
+  ThreadPool(const ThreadPool&);
```
这部分要删掉

ForestLH commented:
> 

这个noncopyable的拷贝构造函数和=函数，应该设置为public吧
```cpp
 private:
  noncopyable(const noncopyable&) = delete;
  void operator=(const noncopyable&) = delete;
```

ForestLH committed:
style:format codes

remove redundant extern g_pika_conf statement and correct if statement judgement instead of assignment

Signed-off-by: Hao Lee <1838249551@qq.com>

ForestLH committed:
Merge pull request #6 from ForestLH/style/clang

style:format codes

AlexStocks commented:
> noncopyable

你看下 https://www.boost.org/doc/libs/1_64_0/boost/core/noncopyable.hpp

AlexStocks committed:
format codes

AlexStocks committed:
merge upstream unstable

AlexStocks committed:
Merge branch 'unstable' into lihao-clang

ForestLH committed:
Merge pull request #7 from AlexStocks/lihao-clang

merge upstream unstable and format codes

infdahai commented:
a huge contribution!

AlexStocks committed:
format codes

AlexStocks commented:
> a huge contribution!

haha. I have worked on this pr for this whole week.

infdahai commented:
@ForestLH If you have questions about noncopyable, you can view the template in muduo(base/noncopyable) or the above boost.

If you use private funcs for copy func or public funcs for deleted copy func, you both get the right answer.

ForestLH committed:
Merge pull request #8 from AlexStocks/lihao-clang

format codes again

yaoyinnan reviewed (approved):
LGTM
